### PR TITLE
feat(FN-2309): handle reconciled records with no payments

### PR DESCRIPTION
**Pull request by RLavender98**
## Introduction :pencil2:
If a fee record has no payments it can be reconciled but have no reconciled by user:
- If the fee record belongs to a manually reconciled report from before the payment reconciliation feature is enabled
- If the fee record is auto reconciled
In both cases we won't actually display any reconciled by data since that is only displayed on the payments tab, so we just need to handle this gracefully.

## Resolution :heavy_check_mark:
- Do not throw an error if a reconciled fee record with no payments has no reconciled by date or user

## Miscellaneous :heavy_plus_sign:
- Update the sql db seeder to not seed reconciliation data for the manually reconciled reports fee records

### CODE REVIEW

**Review by sw-mattw — DISMISSED** [review]

File: dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts [hunk 1]
```diff
@@ -23,14 +23,17 @@ export const getFeeRecordPaymentEntityGroupReconciliationData = async (
 
   const reconciledFeeRecords = group.feeRecords.filter(({ status }) => status === FEE_RECORD_STATUS.RECONCILED);
 
-  if (reconciledFeeRecords.length === 0) {
+  if (reconciledFeeRecords.length !== group.feeRecords.length) {
```
Comment (ttbarnes) [inline dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts:26]: Can we add a comment/documentation here explaining why we need to do this?

Comment (sw-mattw) [reply dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts]: Just picked up this PR, I believe this is to ensure we only return the reconciliation data if all fee records in the group are reconciled. I've pulled this condition out into a self-describing constant to try and avoid the comment - is this ok? Thanks

Comment (sw-mattw) [reply dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts]: This also means we can simplify the filtering to be an `every` check (see diff), simplifying the function

**Review by ttbarnes — APPROVED** [review]

File: dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts (hunk 1 above)
Comment (ttbarnes) [reply dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts]: Nice!

**Pull request merged**